### PR TITLE
KOA-5560 Update chip to semantic colours

### PR DESCRIPTION
**Pull request by marianeum**
Also removed the deprecated custom properties & outline chip following iOS's decision

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/chip/BpkChip.kt [hunk 1]
```diff
@@ -67,19 +70,27 @@ fun BpkChip(
...
+  val interactionSource = remember { MutableInteractionSource() }
+
   val backgroundColor by animateColorAsState(
     targetValue = when {
-      !enabled -> dynamicColorOf(BpkColor.SkyGrayTint06, BpkColor.BlackTint03)
-      selected -> BpkTheme.colors.primary
-      else -> style.backgroundColor
+      !enabled -> BpkChipColors.disabledBackground
+      selected -> style.selectedBackgroundColor
+      else -> interactionSource.animateAsColor(
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/chip/BpkChip.kt:79]: it's by design that selected doesn't have a pressed state

File: Backpack/src/androidTest/java/net/skyscanner/backpack/chip/BpkChipTest.kt [hunk 2]
```diff
@@ -43,42 +42,17 @@ class BpkChipTest {
     context = activityRule.activity
   }
 
-  @Test
-  fun test_message() {
```
Comment (marianeum) [inline Backpack/src/androidTest/java/net/skyscanner/backpack/chip/BpkChipTest.kt:47]: these are things covered by the screenshot tests

File: Backpack/src/main/java/net/skyscanner/backpack/chip/BpkChip.kt [hunk 3]
```diff
@@ -47,38 +47,6 @@ open class BpkChip @JvmOverloads constructor(
 
   private var appearance: BpkChipAppearance
 
-  @Deprecated("Custom background colour is no longer supported. Use existing styles")
```
Comment (marianeum) [inline Backpack/src/main/java/net/skyscanner/backpack/chip/BpkChip.kt:50]: aligning with iOS to remove those

File: Backpack/src/main/java/net/skyscanner/backpack/chip/internal/BpkChipAppearances.kt [hunk 4]
```diff
@@ -31,52 +31,6 @@ import net.skyscanner.backpack.util.use
 
 internal sealed class BpkChipAppearances : BpkChipAppearance {
 
-  internal class Stroke(
```
Comment (marianeum) [inline Backpack/src/main/java/net/skyscanner/backpack/chip/internal/BpkChipAppearances.kt:34]: and this

File: Backpack/src/main/java/net/skyscanner/backpack/chip/internal/Utils.kt [hunk 5]
```diff
@@ -40,16 +40,19 @@ internal fun chipRoundedRect(
...
   selected: Int,
   default: Int,
+  pressed: Int,
   disabled: Int,
 ) = ColorStateList(
   arrayOf(
     intArrayOf(android.R.attr.state_enabled, android.R.attr.state_selected),
-    intArrayOf(android.R.attr.state_enabled),
     intArrayOf(-android.R.attr.state_enabled),
+    intArrayOf(android.R.attr.state_pressed),
```
Comment (marianeum) [inline Backpack/src/main/java/net/skyscanner/backpack/chip/internal/Utils.kt:49]: again the order is intentional here - pressed only applies to the non-selected & enabled state

File: app/src/androidTest/java/net/skyscanner/backpack/chip/BpkChipTest.kt [hunk 6]
```diff
@@ -155,4 +139,13 @@ class BpkChipTest : BpkSnapshotTest() {
     view.disabled = true
     snap(view.wrapInParent())
   }
+
+  private fun BpkChip.wrapInParent(): FrameLayout {
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/chip/BpkChipTest.kt:143]: moved - we no longer need the util class as there's only one chip test class

**Review by bvitaliyg — COMMENTED** [review]
Just a question

File: Backpack/src/main/java/net/skyscanner/backpack/chip/BpkChip.kt [hunk 7]
```diff
@@ -47,38 +47,6 @@ open class BpkChip @JvmOverloads constructor(
...
-  var chipBackgroundColor: Int
```
[the remark this answers is not in the document]
Comment (marianeum) [reply Backpack/src/main/java/net/skyscanner/backpack/chip/BpkChip.kt:51]: Yes in combination with outline buttons - I already communicated the removal to wasabi who owns the usage. Taking the same approach as iOS to remove these

**Review by bvitaliyg — APPROVED** [review]

**Pull request merged**